### PR TITLE
Use fullPathResolver for viewing local files

### PR DESCRIPTION
**Pull request by gerhardol**
Part of #9213

## Proposed changes

Refactor: Use existing FullPathResolve instead of multiple invocations.
This removes a File.Exists() that is resolved incorrectly.
If opening GE with openrepo arg (and more?), open another and viewing a new file in Worktree commit that exist in the first repo, the file in the existing repo was shown.

## Test methodology 

Manual

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by mstv — APPROVED** [review]
LGTM

File: GitUI/Editor/FileViewer.cs [hunk 1]
```diff
@@ -607,14 +607,12 @@ public Task ViewFileAsync(string fileName, bool isSubmodule = false, Action? ope
             {
                 try
                 {
-                    var path = _fullPathResolver.Resolve(fileName);
-
-                    if (!File.Exists(path))
+                    if (!File.Exists(fullPath))
```
Comment (RussKie) [inline GitUI/Editor/FileViewer.cs:610]: This should be the way around - the path must be resolved via `_fullPathResolver`, i.e. resolved for the _current_ working directory, and not some arbitrary path.

**Review by RussKie — CHANGES_REQUESTED** [review]

File: GitUI/Editor/FileViewer.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/Editor/FileViewer.cs:610]: The full path is already resolved, no need to do it again

**Comment by gerhardol** [comment]
Rebased without changes

This solves what is reported in #9213.
I have investigated som, not found similar issues
But also not found why this occurs when starting with command line arguments

File: GitUI/Editor/FileViewer.cs (hunk 1 above)
Comment (RussKie) [inline GitUI/Editor/FileViewer.cs:610]: Let's add an assert
```suggestion
                    Debug.Assert(Path.IsPathFullyQualified(path), "Path must be resolved and fully qualified");
                    if (!File.Exists(fullPath))
```

File: GitUI/Editor/FileViewer.cs [hunk 2]
```diff
@@ -625,16 +623,12 @@ public Task ViewFileAsync(string fileName, bool isSubmodule = false, Action? ope
 
             string GetFileText()
             {
-                var path = File.Exists(fileName)
-                    ? fileName
-                    : _fullPathResolver.Resolve(fileName);
-
-                if (!File.Exists(path))
+                if (!File.Exists(fullPath))
```
Comment (RussKie) [inline GitUI/Editor/FileViewer.cs:626]: ditto here

File: GitUI/Editor/FileViewer.cs [hunk 3]
```diff
@@ -843,25 +837,22 @@ private void OnExtraDiffArgumentsChanged()
...
             return ShowOrDeferAsync(GetFileLength(), showFunc);
 
             long GetFileLength()
             {
                 try
                 {
-                    var resolvedPath = _fullPathResolver.Resolve(fileName);
-
-                    if (File.Exists(resolvedPath))
+                    if (File.Exists(fullPath))
```
Comment (RussKie) [inline GitUI/Editor/FileViewer.cs:848]: ditto here

**Review by RussKie — APPROVED** [review]
Tested locally - 👍

File: GitUI/Editor/FileViewer.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/Editor/FileViewer.cs:610]:
```suggestion
                    if (!File.Exists(fullPath))
                    Debug.Assert(Path.IsPathFullyQualified(fullPath), "Path must be resolved and fully qualified");
```

File: GitUI/Editor/FileViewer.cs (hunk 2 above)
Comment (gerhardol) [reply GitUI/Editor/FileViewer.cs:626]: Refactoring the code instead.

File: GitUI/Editor/FileViewer.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/Editor/FileViewer.cs:610]: No, we want to assert before checking file.

Comment (gerhardol) [reply GitUI/Editor/FileViewer.cs]: Made a minor refactoring of the handling with properly ordered checks (a few lines removed)

File: GitUI/Editor/FileViewer.cs [hunk 4]
```diff
@@ -575,26 +576,32 @@ string GetFileTextIfBlobExists()
...
         {
             string? fullPath = _fullPathResolver.Resolve(fileName);
-
-            if (isSubmodule && !GitModule.IsValidGitWorkingDir(fullPath))
-            {
-                return ViewTextAsync(fileName, "Invalid submodule: " + fileName);
-            }
-
             Validates.NotNull(fullPath);
+            Debug.Assert(Path.IsPathFullyQualified(fullPath), "Path must be resolved and fully qualified");
```
Comment (mstv) [inline GitUI/Editor/FileViewer.cs:580]: Is `fullPath` really not `null` here now? Then its type could be (non-nullable) `string`.

Comment (gerhardol) [reply GitUI/Editor/FileViewer.cs:580]: _fullPathResolver.Resolve() is string?, but the result should not be null in our case (and null was actually OK).
ValidatesNotNull() must check the result, to follow the contract of other methods.

**Comment by gerhardol** [comment]
@RussKie Do you want to review the latest changes (after your comments) or can I merge?
You set the 3.5.1 MS too, need to reformat for 3.5.

**Comment by RussKie** [comment]
Yep, thank you.
I think it's worth servicing in 3.5

**Pull request merged**